editor/editor_themes.cpp

```diff
@@ -948,8 +960,11 @@ Ref<Theme> create_editor_theme(const Ref<Theme> p_theme) {
 
 	// Tabs & TabContainer
 	theme->set_stylebox("tab_selected", "TabContainer", style_tab_selected);
+	theme->set_stylebox("tab_selected_bottom", "TabContainer", style_tab_selected_bottom);
 	theme->set_stylebox("tab_unselected", "TabContainer", style_tab_unselected);
+	theme->set_stylebox("tab_unselected_bottom", "TabContainer", style_tab_unselected_bottom);
 	theme->set_stylebox("tab_disabled", "TabContainer", style_tab_disabled);
+	theme->set_stylebox("tab_disabled_bottom", "TabContainer", style_tab_disabled_bottom);
```

Comment:
I'm thinking, maybe we could somehow flip the existing styleboxes inside of the TabContainer instead of creating a second set of mirrored styles? Can probably even add it as a method of each StyleBox type for better encapsulation.

Managing two sets of theme items is bothersome.

Reply:
Sounds good, but it also sounds like a lot of work. Currently there is no API to flip or rotate textures, which you would need for `StyleBoxTexture`. Easy enough for `StyleBoxFlat` I guess, but:
* Did we only want to do it for one of the stylebox classes?
* Is it a blocking issue for this feature?

Reply:
You can rotate the Image. So something like `texture->get_image()` -> `flip_v` -> `ImageTexture.new()`.

Reply:
Ah my bad, thanks. Ok, I could look into that.

Reply:
> Is it a blocking issue for this feature?

To address this question, yes, I'd like to decide on that now before we merge, as this PR adds stuff and removing it later is going to be harder than not adding it in the first place 🙂 As I don't see why a user would want to have both upwards and downwards styleboxes at the same time on the same control, I think duplication should be avoided.

Reply:
Won't this run into issues since the `texture` in `StyleBoxTexture` can be any `texture`? So you can't just create an `ImageTexture` because the original texture may have been `AnimatedTexture`. Also doing `flip` iterates over every pixel, so I am caching the results as if not, it could be running almost every frame.

Reply:
Caching is okay, it's done often for theme items. But you are right, some types of styleboxes are harder to handle than others here. Well, since the editor relies on the StyleBoxFlat we could maybe just flip them, and add them as a type variation? 🤔 I would still avoid duplication here.

Reply:
The default theme uses style box texture though. I don't think we can leave that one looking weird.

Reply:
I think Calinou has a PR to change it to flats for 4.0.

Reply:
Still though, if someone uses style box texture in their tab container, and then they want to use the tabs at bottom feature, it won't work...

Reply:
If the default will work properly, theme variation for custom styles is fine.

Or if we go for auto-flipping, we could just make it support only some texture types (and possibly be optional).

Reply:
If someone creates a custom theme, they can customize type variation as well. In practice it would be similar to how it is now, but without duplication and with the explicit setting to use flipped styles alongside the flipped tabbar setting.

Maybe not the most obvious flow, but we can work on discoverability. I think type variations are a better way to handle permutations than additional style for each main style.